gax-java/gax/src/main/java/com/google/api/gax/rpc/TransportChannelProvider.java

```diff
@@ -97,6 +97,9 @@ public interface TransportChannelProvider {
    */
   TransportChannelProvider withEndpoint(String endpoint);
 
+  /** Sets whether to use S2A when constructing a new {@link TransportChannel}. */
+  TransportChannelProvider withUseS2A(boolean useS2A);
```

Comment:
Can we make this a default method and throw `UnsupportedOperationException`? So that we don't have to implement every subclasses for this interface. Even though this is an internal class, it also prevents customers from breaking changes if they use it accidentally.

Reply:
Doesn't this cause a problem when from ClientContext we call:
`transportChannelProvider.withUseS2A()`, where `transportChannelProvider` is an instance of `TransportChannelProvider`? In this case it would throw an `UnsupportedOperationException` even though `transportChannelProvider` could be a `InstantiatingGrpcChannelProvider`. We could add a check and cast `transportChannelProvider` to an instance of `InstantiatingGrpcChannelProvider` but then gax would have to have a dependency on gax-grpc (which would be a circular dependency I think). Let me know if I'm misunderstanding something here.

Reply:
> Doesn't this cause a problem when from ClientContext

Yes, it does. Sorry if I was not clear, what I was trying to get to is that `useS2A` would not be used in all other implementations, including `FixedTransportChannelProvider`, `InstantiatingHttpJsonChannelProvider` and possible customers' own implementation. So we should 
1. Make this a default method and throw `UnsupportedOperationException`, so that customers would not get breaking changes if they implemented this interface.
2. Implement this method in `FixedTransportChannelProvider` and `InstantiatingHttpJsonChannelProvider` with no-op, probably just returning `this` should be good enough because they are not used right now. We would probably never use `useS2A` in `FixedTransportChannelProvider`, and we should only implement it in `InstantiatingHttpJsonChannelProvider` when we are ready to support it in HttpJson transport.